oxidation/parsec_cli/parsec_sequestre_dump/src/model.rs

```diff
@@ -0,0 +1,519 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+use std::error::Error;
+
+use diesel::{ExpressionMethods, QueryDsl, RunQueryDsl, SqliteConnection};
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+use crate::schema::{block, info, vlob_atom};
+use parsec_api_crypto::SecretKey;
+use parsec_api_types::{DeviceID, FileManifest, FolderManifest, Manifest, WorkspaceManifest};
+
+type Time = chrono::DateTime<chrono::Utc>;
+
+#[cfg(test)]
+use crate::schema::{device, realm_role, user_ as user};
+#[cfg(test)]
+use tests_fixtures::Device;
+
+#[derive(Debug, Queryable, Serialize, Deserialize, PartialEq)]
+pub struct Block {
+    pub block_id: Vec<u8>,
+    pub data: Vec<u8>,
+    pub author: i32,
+    pub size: i32,
+    pub created_on: f64,
+}
+
+#[derive(Debug, Queryable, Serialize, Deserialize, PartialEq)]
+pub struct VlobAtom {
+    pub vlob_id: Vec<u8>,
+    pub version: i32,
+    pub blob: Vec<u8>,
+    pub size: i32,
+    pub author: i32,
+    pub timestamp: f64,
+}
+
+#[derive(Debug, Queryable, Serialize, Deserialize, PartialEq)]
+pub struct Info {
+    pub magic: i32,
+    pub version: i32,
+    pub realm_id: Vec<u8>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct Data {
+    #[serde(default, skip_serializing_if = "String::is_empty")]
+    pub name: String,
+    pub author: DeviceID,
+    pub timestamp: Time,
+    pub id: Uuid,
+    pub version: u32,
+    pub created: Time,
+    pub updated: Time,
+    pub size: u64,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub children: Vec<Data>,
+    #[serde(default, skip_serializing_if = "String::is_empty")]
+    pub content: String,
+}
+
+impl std::fmt::Debug for Data {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Data")
+            .field("name", &self.name)
+            .field("children", &self.children)
+            .finish()
+    }
+}
+
+impl Default for Data {
+    fn default() -> Self {
```

Comment:
We never create Data object in this code (since it always code from the database)

I guess you only need this in the tests, if so would it be possible to instead define a `generate_empty_data()` function in the tests ?

Reply:
No need anymore